examples/inference-deployments/mpt/mpt_7b_ft_handler.py

```diff
@@ -65,15 +65,21 @@ def __init__(self,
         self.device = torch.cuda.current_device()
         self.model_name = model_name
 
-        # Datatype of weights in the HF checkpoint
-        weight_data_type = 'fp32'
-        convert_mpt_to_ft(self.model_name, LOCAL_CHECKPOINT_PATH, gpus,
-                          weight_data_type, False)
-        ckpt_config = configparser.ConfigParser()
         model_path = os.path.join(LOCAL_CHECKPOINT_PATH, f'{gpus}-gpu')
         ckpt_config_path = os.path.join(model_path, 'config.ini')
-        if os.path.isfile(ckpt_config_path):
-            ckpt_config.read(ckpt_config_path)
+        # If FT checkpoint doesn't exist, create it.
+        if not os.path.isfile(ckpt_config_path):
+            print('Converting model to FT format')
+            # Datatype of weights in the HF checkpoint
+            weight_data_type = 'fp32'
+            convert_mpt_to_ft(self.model_name, LOCAL_CHECKPOINT_PATH, gpus,
+                              weight_data_type, False)
+            if not os.path.isfile(ckpt_config_path):
+                raise RuntimeError('Failed to create FT checkpoint')
+        else:
+            print(f'Reusing existing FT checkpoint at {model_path}')
+        ckpt_config = configparser.ConfigParser()
+        ckpt_config.read(ckpt_config_path)
```

Comment:
I think it might be cleaner to move this into a custom download function (like `download_convert_mpt_ft` and then set the `downloader` path in the input yaml to `examples.inference-deployments....download_convert_mpt_ft` -- this gets run first thing when the pod spins up, so we can fail early if there's any issue with the checkpoint conversion

Reply:
yeah that sounds like a good idea. will use a follow-up PR for that.